apps/api/src/app/environments/usecases/delete-environment/delete-environment.usecase.ts

```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@nestjs/common';
+import { EnvironmentEntity, EnvironmentRepository } from '@novu/dal';
+
+import { DeleteEnvironmentCommand } from './delete-environment.command';
+
+@Injectable()
+export class DeleteEnvironment {
+  constructor(private environmentRepository: EnvironmentRepository) {}
+
+  async execute(command: DeleteEnvironmentCommand) {
+    return await this.environmentRepository.delete({
+      _id: command._id,
+    });
+  }
+}
```

Comment:
The delete enedpoint might create quite a few un predictable behaviours. So I would prefer if we wouldn't add this functionality to the API at the moment.